### ISSUE TITLE
Catching other URLS

### ISSUE THREAD

**Issue by mitra42** [abhidas17695/dweb-ext#31]
Now that we've done this, we can intercept other dweb style URLs.

I'm not sure what the rules are, can we for example intercept any or all of:
dweb:*
ipfs:*
magnet:*

**Comment by abhidas17695** [comment]
We could catch all such URLs with the code `url.startsWith('dweb') || url.startsWith('ipfs') || url.startsWith('magnet')`
Did you have this in mind ?

**Comment by mitra42** [comment]
Great - no change is required yet, I need to figure out what I want to do with them first.

**Comment by mitra42** [comment]
At the Dweb summit I spoke to people from IPFS, Mozilla and GUN to come to some agreement on how to better support URIs across platforms. The draft of my documentation is at [https://github.com/mitra42/dweb-universal/blob/master/naming.md] .

The next step for dweb-ext would be I believe to: 
* recognize the patterns described in that document. 
* Canonicalize them to /proto/internal for example /ipfs/Q1234/a/b/c  this should be a function in Domain.js so it can be used more widely than in dweb-ext
* Pass the canonicalized name to Domain.resolve (via rootResolve or resolveAndBoot )
* Add leaf records to Domain._setupOnce for /gun and /ipfs and /magnet, this should be straightforward but might require hacking leaf records as it may extend what they were intended for. 
* Possibly add a hard-coded set of top level Domains under the root to speed things up. 

I'm happy to chat about this to clear up any questions, note that I'm on Australia (Sydney) time currently which is 4.5 hours ahead of India.

**Comment by abhidas17695** [comment]
I'l check out the document today. Will you be available on Slack ?

**Comment by mitra42** [comment]
Yes - but only for an hour or so, it might be better tomorrow whenever you are available.
